### PR TITLE
[stable/redis-ha] add persistentVolume.reclaimPolicy

### PR DESCRIPTION
**Pull request by morganchristiansson**
also document extra persistentVolume variables and reformat markdown table

To hide whitespace changes in README.md press the cog and tick _Hide whitespace changes_  or add `&w=1` to the query parameters.

#### What this PR does / why we need it:

Adds ability to set persistentVolumeReclaimPolicy so that persistent volumes are deleted with the statefulset and not left behind.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/redis-ha/README.md [hunk 1]
```diff
@@ -82,55 +82,58 @@ The following table lists the configurable parameters of the Redis chart and the
...
+| `persistentVolume.size`          | Size for the volume                                                                                                                                                                               | 10Gi                                                                                       |
+| `persistentVolume.annotations`   | Annotations for the volume                                                                                                                                                                        | `{}`                                                                                       |
+| `persistentVolume.reclaimPolicy` | Method used to reclaim an obsoleted volume. `Delete` or `Retain`                                                                                                                                  | `Retain`                                                                                   |
```
Comment (DandyDeveloper) [inline stable/redis-ha/README.md]: Default should be `nil` as you've commented out the section in the file.

File: stable/redis-ha/values.yaml [hunk 2]
```diff
@@ -301,6 +301,8 @@ persistentVolume:
     - ReadWriteOnce
   size: 10Gi
   annotations: {}
+  # Set reclaimPolicy to Delete to delete obsoleted volumes
```
Comment (DandyDeveloper) [inline stable/redis-ha/values.yaml]:
```suggestion
  # reclaimPolicy per https://kubernetes.io/docs/concepts/storage/persistent-volumes/#reclaiming
```

File: stable/redis-ha/values.yaml [hunk 3]
```diff
@@ -301,6 +301,8 @@ persistentVolume:
...
+  # reclaimPolicy: Retain
```
Comment (DandyDeveloper) [inline stable/redis-ha/values.yaml]:
```suggestion
  reclaimPolicy: ""
```

**Review by DandyDeveloper — CHANGES_REQUESTED** [review]
Better to have it nil like this and have doc references for those not sure about it.

Minor changes but looks good.

File: stable/redis-ha/README.md (hunk 1 above)
Comment (morganchristiansson) [reply stable/redis-ha/README.md]: Retain is already the default in k8s. But this is cleaner - thanks.

**Comment by morganchristiansson** [comment]
Added commit to implement feedback from @DandyDeveloper - change persistentVolume.reclaimPolicy default to nil

rebased on latest and force-pushed

**Comment by shabbir-taulia** [comment]
Sorry for commenting on a closed PR but it looks like the `ReclaimPolicy` is still not supported for dynamically provisioned volumes. [upstream issue](https://github.com/kubernetes/kubernetes/issues/30606)

From the docs: https://kubernetes.io/docs/tasks/administer-cluster/change-pv-reclaim-policy/

```
For dynamically provisioned PersistentVolumes, the default reclaim policy is "Delete". This means that a dynamically provisioned volume is automatically deleted when a user deletes the corresponding PersistentVolumeClaim. This automatic behavior might be inappropriate if the volume contains precious data
```

I might be wrong but validation fails with the following error
```
➜ (⎈ gke_ops_us-central1_cd:default) tmp ✗ k apply --validate=true --dry-run=true -f output/argo-cd/charts/argo-cd/charts/redis-ha/templates/redis-ha-statefulset.yaml
error: error validating "output/argo-cd/charts/argo-cd/charts/redis-ha/templates/redis-ha-statefulset.yaml": error validating data: ValidationError(StatefulSet.spec.volumeClaimTemplates[0].spec): unknown field "persistentVolumeReclaimPolicy" in io.k8s.api.core.v1.PersistentVolumeClaimSpec; if you choose to ignore these errors, turn validation off with --validate=false

**Pull request merged**